R/git.R

```diff
@@ -453,51 +453,141 @@ use_git_credentials <- function(credentials) {
-  kv_line("Name", name)
-  kv_line("Email", email)
-  kv_line("Default protocol", getOption("usethis.protocol"))
+  # git user ------------------------------------------------------------------
+  hd_line("Git user")
+  kv_line("Name", git_config_get("user.name", global = TRUE))
+  kv_line("Email", git_config_get("user.email", global = TRUE))
   ## TODO: forward info from the credentials package once we start using it
+  ## and it reflects the credentials situation usethis will actually meet
   ## e.g., git version, HTTPS credential helpers, SSH keys, etc.
   kv_line("Vaccinated", git_vaccinated())
 
-  hd_line("git2r")
-  kv_line("Supports SSH", git2r::libgit2_features()$ssh)
+  # usethis + git2r ----------------------------------------------------------
+  hd_line("usethis + git2r")
+  kv_line("Default usethis protocol", getOption("usethis.protocol"))
+  kv_line("git2r supports SSH", git2r::libgit2_features()$ssh)
   credentials_value <- if (have_git2r_credentials()) {
     glue("<user-provided git2r credential object with class {class(git_credentials())}>")
   } else {
     "<usethis + git2r default behaviour>"
   }
   kv_line("Credentials", credentials_value)
 
-  if (proj_active()) {
-    hd_line("Project")
-    if (uses_git()) {
-      repo <- git_repo()
-      kv_line("Path", repo$path)
-      kv_line("Local branch", git_branch_name())
-      kv_line("Remote branch", git_branch_tracking())
-    } else {
-      cat_line("Git not activated")
-    }
-  }
-
+  # github user ---------------------------------------------------------------
   hd_line("GitHub")
-  if (have_github_token()) {
-    ui_done("Personal access token found in env var")
+  have_token <- have_github_token()
+  if (have_token) {
+    kv_line("Personal access token", "<found in env var>")
     who <- github_user()
     if (is.null(who)) {
-      cat_line("Token is invalid")
+      ui_oops("Token is invalid.")
+      have_token <- FALSE
     } else {
       kv_line("User", who$login)
       kv_line("Name", who$name)
     }
   } else {
-    cat_line("No personal access token found")
+    kv_line("Personal access token", NULL)
+  }
+
+  # repo overview -------------------------------------------------------------
+  hd_line("Repo")
+  withr::with_options(
+    list(usethis.quiet = TRUE),
+    try(proj_get(), silent = TRUE)
+  )
+  if (!proj_active()) {
+    ui_info("No active usethis project.")
+    return(invisible())
+  }
+
+  if (!uses_git()) {
+    ui_info("Active project is not a Git repo.")
+    return(invisible())
+  }
+
+  kv_line("Path", git_repo()$path)
+  branch <- tryCatch(git_branch_name(), error = function(e) NULL)
+  tracking_branch <- if (is.null(branch)) NULL else git_branch_tracking()
+  ## TODO: rework when ui_*() functions make it possible to do better
+  branch <- if (is.null(branch)) "<unset>" else branch
+  tracking_branch <- if (is.null(tracking_branch)) "<unset>" else tracking_branch
+  cat_line("* ", "Local branch -> remote tracking branch: ",
```

Comment:
Needs style tweak

Reply:
Yes, see that's what my comment and TODO in the code are about. I'm trying not to put any `ui.R` changes in this PR.

Reply:
I meant code style - the indenting is the tidyverse standard.